src/me/StevenLawson/TotalFreedomMod/Commands/Command_nick.java

```diff
@@ -0,0 +1,72 @@
+package me.StevenLawson.TotalFreedomMod.Commands;
+
+import me.StevenLawson.TotalFreedomMod.Bridge.TFM_EssentialsBridge;
+import me.StevenLawson.TotalFreedomMod.TFM_Util;
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+
+@CommandPermissions(level = AdminLevel.OP, source = SourceType.ONLY_IN_GAME)
+@CommandParameters(description = "Give yourself a nickname.", usage = "/<command> <nickname>", aliases = "nickname")
+public class Command_nick extends TFM_Command
+{
+    @Override
+    public boolean run(CommandSender sender, Player sender_p, Command cmd, String commandLabel, String[] args, boolean senderIsConsole)
+    {
+        if (args.length < 1)
+        {
+            playerMsg(ChatColor.RED + "You didn't provide a nickname!");
+        }
+        if (args.length > 1)
+        {
+            playerMsg(ChatColor.RED + "Too many command arguments!");
+        }
+        if ("off".equals(args[0]))
+        {
+            TFM_EssentialsBridge.setNickname(sender.getName(), null);
+            playerMsg(ChatColor.GOLD + "You no longer have a nickname.");
+            return true;
+        }        
+        final String nickPlain = ChatColor.stripColor(TFM_Util.colorize(args[0].trim()));
+        final String nickInput = args[0];
+
+        if (!nickPlain.matches("^[a-zA-Z_0-9\u00a7]+$"))
+        {
+            playerMsg(ChatColor.RED + "Your nickname contains invalid characters.");
+            return true;
+        }
+        if (nickPlain.length() < 4 || nickPlain.length() > 30)
+        {
+            playerMsg("Your nickname must be between 4 and 30 characters long.");
+            return true;
+        }
+        if (nickInput.contains("&k") || (nickInput.contains("&m") || (nickInput.contains("&n") || (nickInput.contains("&o")) || nickInput.contains("&0"))))
```

Comment:
A method in TFM_Util would be nice. Also, this doesn't catch the actual paragraph characters. So \u00A7C would still be in Red.